ksp_plugin/flight_plan.cpp

```diff
@@ -265,6 +292,31 @@ void FlightPlan::PopLastSegment() {
   CHECK(!trajectory->is_root());
   trajectory->parent()->DeleteFork(&trajectory);
   segments_.pop_back();
+  if (anomalous_segments_ > 0) {
+    --anomalous_segments_;
+  }
+}
+
+bool FlightPlan::CoastReachesManœuvreInitialTime(
+    DiscreteTrajectory<Barycentric>& coast,
+    NavigationManœuvre const& manœuvre) {
+  // TODO(egg): whenever we call this function we end up recomputing the
+  // following trajectory in |Append|; we could pass it to |Append| instead,
+  // but then we would have to deal with the |set_on_destroy|.
+  DiscreteTrajectory<Barycentric>* recomputed_coast =
```

Comment:
Why does this have to be a fork and not a separate trajectory?  It seems to me that `coast` is not modified but needs to be non-const because of this forking.